src/Handler/StreamHandler.php

```diff
@@ -106,18 +107,30 @@ private function checkDecode(array $options, array $headers, $stream)
     {
         // Automatically decode responses when instructed.
         if (!empty($options['decode_content'])) {
-            foreach ($headers as $key => $value) {
-                if (strtolower($key) == 'content-encoding') {
-                    if ($value[0] == 'gzip' || $value[0] == 'deflate') {
-                        return new Psr7\InflateStream(
-                            Psr7\stream_for($stream)
-                        );
+            $normalizedKeys = array_combine(array_map('strtolower', array_keys($headers)), array_keys($headers));
```

Comment:
Can you modify this to use `array_change_key_case` here and in the curl handler code? http://php.net/manual/en/function.array-change-key-case.php